### PR TITLE
Feature/conditional-links

### PR DESCRIPTION
**Pull request by natyork**
Render links requiring authorization conditionally per https://www.pivotaltracker.com/story/show/142421857

Note, there are currently no policies applied to PeopleController, therefore the links on People#index and People#show have not been updated. I have created a new story in the icebox to update the authorizations for PeopleController, including conditionally rendering links on the associated views (https://www.pivotaltracker.com/story/show/142992497)

### CODE REVIEW

File: app/views/events/_event.html.erb [hunk 1]
```diff
@@ -2,7 +2,9 @@
   <td><%= event.name %></td>
   <td>
     <%= link_to 'Show', event_path(event) %>
-    <%= link_to 'Edit', edit_event_path(event) %>
-    <%= render partial: 'delete-link', locals: { event: event } %>
+    <% if current_person&.admin? %>
```
Comment (graygilmore) [inline app/views/events/_event.html.erb]: Should this use the event policy instead?

Comment (natyork) [reply app/views/events/_event.html.erb]: Yes that makes sense - currently only an admin is permitted to edit/delete events, but if that changes in future, we wouldn't have to update the view.

**Comment by natyork** [comment]
Updated per @graygilmore comments

File: app/views/events/_event.html.erb [hunk 2]
```diff
@@ -2,7 +2,11 @@
...
   <td>
     <%= link_to 'Show', event_path(event) %>
-    <%= link_to 'Edit', edit_event_path(event) %>
-    <%= render partial: 'delete-link', locals: { event: event } %>
+    <% if policy(event).edit? %>
+      <%= link_to 'Edit', edit_event_path(event) %>
+    <% end %>
+    <% if policy(event).destroy? %>
+      <%= render partial: 'delete-link', locals: { event: event } %>
+    <%end%>
```
Comment (jacquesporveau) [inline app/views/events/_event.html.erb]: Spaces

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -10,5 +10,9 @@
   <dd><%= @event.location %></dd>
 </dl>
 
-<%= link_to 'Edit', edit_event_path(@event) %>
-<%= render partial: 'delete-link', locals: { event: @event } %>
+<% if policy(@event).edit? %>
+  <%= link_to 'Edit', edit_event_path(@event) %>
+<% end %>
```
Comment (jacquesporveau) [inline app/views/events/show.html.erb]: `<%= link_to 'Edit', edit_event_path(@event) if policy(@event).edit? %>`

Could probably one line these. That one is 70 char.

**Review by jacquesporveau — CHANGES_REQUESTED** [review]
The spacing issue is required but the one line if statement is a preference thing that I'll leave up to you.

**Comment by natyork** [comment]
Updated per @jacquesporveau comments

File: app/views/events/_event.html.erb [hunk 4]
```diff
@@ -2,7 +2,10 @@
   <td><%= event.name %></td>
   <td>
     <%= link_to 'Show', event_path(event) %>
-    <%= link_to 'Edit', edit_event_path(event) %>
-    <%= render partial: 'delete-link', locals: { event: event } %>
+    <%= link_to('Edit', edit_event_path(event)) if policy(event).edit? %>
+
+    <% if policy(event).destroy? %>
+      <%= render partial: 'delete-link', locals: { event: event } %>
+    <%end%>
```
Comment (jacquesporveau) [inline app/views/events/_event.html.erb]: Expand your brackets... expand your life
`<% end %>`

**Review by jacquesporveau — CHANGES_REQUESTED** [review]

**Comment by natyork** [comment]
Updated per @jacquesporveau comment

**Review by qr8r — APPROVED** [review]

**Review by jacquesporveau — APPROVED** [review]

**Pull request merged**